_posts/2019-05-24-varying-slopes/varying_slopes.Rmd

```diff
@@ -0,0 +1,748 @@
+---
+title: "Hierarchical partial pooling, continued: Varying slopes models with TensorFlow Probability"
+description: > 
+   This post builds on our recent introduction to multi-level modeling with tfprobability, the R wrapper to TensorFlow Probability. We show how to pool not just mean values ("intercepts"), but also relationships ("slopes"), thus enabling models to learn from data in an even broader way. Again, we use an example from Richard McElreath's "Statistical Rethinking"; and again, terminology as well as the way we present this topic are largely owed to this book.
+author:
+  - name: Sigrid Keydana
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: keydana2019cafes
+date: 05-24-2019
+categories:
+  - Probability and statistics 
+  - Introductions
+output:
+  distill::distill_article:
+    self_contained: false
+preview: images/thumb.png
+---
+
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+
+In a previous post, we showed how to use [tfprobability](https://rstudio.github.io/tfprobability/) -- the R interface to TensorFlow Probability -- to build a _multilevel_, or  _partial pooling_, model of tadpole survival in differently sized (and thus, differing in inhabitant number) tanks.
+
+A completely _pooled_ model would have resulted in a global estimate of survival count, irrespective of tank; an _unpooled_ model would have learned to predict survival count for each tank separately. The former approach does not take into account different circumstances; the latter does not make use of common information. (Also, it clearly has no predictive use unless we want to make predictions for the very same entities we used to train the model.)
+
+In contrast, a _partially pooled_ model lets you make predictions for the familiar, as well as new entities: Just use the appropriate prior.
+
+Assuming we _are_ in fact interested in the same entities -- why would we want to apply partial pooling?
+For the same reasons so much effort in machine learning goes into devising regularization mechanisms.
+We don't want to overfit too much to actual measurements, be they related to the same entity or a class of entities. If I want to predict my heart rate as I wake up next morning, based on a single measurement I'm taking now (let's say it's evening and I'm frantically typing a blog post), I better take into account some facts about heart rate behavior in general (instead of just projecting into the future the exact value measured right now).
```

Comment:
Have this paragraph with the preceding line.

Reply:
> If I want to predict my heart rate as I wake up next morning, based on a single measurement I'm taking now (let's say it's evening and I'm frantically typing a blog post), I better take into account some facts about heart rate behavior in general (instead of just projecting into the future the exact value measured right now).

I like this example, and I wonder if it would help to add some common terminology in here to help people make connections (e.g. _extrapolation_ etc).

Reply:
1) you mean the 2 single-line ones go together?
2) do you have a suggestion?

Reply:
I kind of disagree with my own comment now, since you expand on it later!

Reply:
haha I'm relieved :-) ... because I don't use "extrapolate", but "project" should convey something similar, right :-)